ibis/backends/tests/test_vectorized_udf.py

```diff
@@ -497,27 +494,33 @@ def test_elementwise_udf_overwrite_destruct_and_assign(
 
 
 @pytest.mark.min_spark_version('3.1')
-def test_elementwise_udf_destruct_exact_once(udf_backend, udf_alltypes):
-    with tempfile.TemporaryDirectory() as tempdir:
-
-        @elementwise(
-            input_type=[dt.double],
-            output_type=dt.Struct(['col1', 'col2'], [dt.double, dt.double]),
-        )
-        def add_one_struct_exact_once(v):
-            key = v.iloc[0]
-            path = Path(f"{tempdir}/{key}")
-            assert not path.exists()
-            path.touch()
-            return v + 1, v + 2
-
-        result = udf_alltypes.mutate(
-            add_one_struct_exact_once(udf_alltypes['index']).destructure()
-        )
+@pytest.mark.parametrize('method', ['destructure', 'unpack'])
+def test_elementwise_udf_destructure_exact_once(
+    udf_backend, udf_alltypes, method, tmp_path
+):
+    @elementwise(
+        input_type=[dt.double],
+        output_type=dt.Struct(['col1', 'col2'], [dt.double, dt.double]),
+    )
+    def add_one_struct_exact_once(v):
+        key = v.iloc[0]
+        path = tmp_path / str(key)
+        assert not path.exists()
+        path.touch()
+        return v + 1, v + 2
+
+    struct = add_one_struct_exact_once(udf_alltypes['index'])
+
+    if method == "destructure":
+        expr = udf_alltypes.mutate(struct.destructure())
+    elif method == "unpack":
+        expr = udf_alltypes.mutate(struct=struct).unpack("struct")
```

Comment:
Does `unpack` does the same thing as `destructure` but a slight different API?

Reply:
Yes. We kept `destructure` to hopefully avoid some breakage.

Reply:
kk, `destructure` isn't directly exposed to our users so it shouldn't be too hard for us to update that API surface.